ceremonies/src/lib.rs

```diff
@@ -359,148 +414,326 @@ decl_error! {
-            <MeetupIndex<T>>::remove_prefix((cid, cindex), None);
             <MeetupCount>::insert((cid, cindex), 0);
             <AttestationRegistry<T>>::remove_prefix((cid, cindex), None);
             <AttestationIndex<T>>::remove_prefix((cid, cindex), None);
             <AttestationCount>::insert((cid, cindex), 0);
             <MeetupParticipantCountVote<T>>::remove_prefix((cid, cindex), None);
+
+            <IssuedRewards>::remove_prefix((cid, cindex), None);
         }
         debug!(target: LOG, "purged registry for ceremony {}", cindex);
     }
 
-    // this function is expensive, so it should later be processed off-chain within SubstraTEE-worker
-    // currently the complexity is O(n) where n is the number of registered participants
-    fn assign_meetups() {
+    fn generate_meetup_assignment_params(community_ceremony: CommunityCeremony) -> DispatchResult {
+        let meetup_multiplier = 10u64;
+        let num_locations = <encointer_communities::Module<T>>::get_locations(&community_ceremony.0).len() as u64;
+        let num_boostrappers = Self::bootstrapper_count(community_ceremony);
+        let num_reputables = Self::reputable_count(community_ceremony);
+        let num_endorsees = Self::endorsee_count(community_ceremony);
+        let num_newbies = Self::newbie_count(community_ceremony);
+
+        if num_locations == 0 {
+            return Err(<Error<T>>::NoLocationsAvailable.into())
+        }
+        let mut num_meetups = min(num_locations, Self::find_prime_below(num_boostrappers + num_reputables));
+
+        let num_assigned_bootstrappers = num_boostrappers;
+
+        let mut available_slots = num_meetups * meetup_multiplier - num_assigned_bootstrappers;
+
+
+        let num_assigned_reputables = min(num_reputables, available_slots);
+        available_slots -= num_assigned_reputables;
+
+        let num_assigned_endorsees = min(num_endorsees, available_slots);
+        available_slots -= num_assigned_endorsees;
+
+        let max_assigned_newbies = (num_assigned_bootstrappers + num_assigned_reputables + num_assigned_endorsees) / 3;
+        let mut num_assigned_newbies = min(num_newbies, available_slots);
+        num_assigned_newbies = min(num_assigned_newbies, max_assigned_newbies);
+
+        let num_participants = num_assigned_bootstrappers + num_assigned_reputables + num_assigned_endorsees + num_assigned_newbies;
+
+        num_meetups = num_participants / meetup_multiplier;
+
+        // ceil
+        if num_participants % meetup_multiplier != 0 {num_meetups += 1;}
+
+
+        let assignment_params_bootstrappers_reputables = Self::generate_assignment_function_params(num_assigned_bootstrappers + num_assigned_reputables, num_meetups);
+        let assignment_params_endorsees = Self::generate_assignment_function_params(num_assigned_endorsees, num_meetups);
+        let assignment_params_newbies = Self::generate_assignment_function_params(num_assigned_newbies, num_meetups);
+
```

Comment:
why introduce variables that are just used once?

Reply:
i think it is more readable like this